modules/nextflow/src/main/groovy/nextflow/script/WaveMetadata.groovy

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2013-2024, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package nextflow.script
+
+import groovy.transform.CompileStatic
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.ToString
+import groovy.util.logging.Slf4j
+import nextflow.Session
+/**
+ * Models Wave metadata for Nextflow execution
+ * 
+ * @author Marco De La Pierre <marco.delapierre@gmail.com>
+ */
+@Slf4j
+@CompileStatic
+@ToString(includeNames = true)
+@EqualsAndHashCode
+class WaveMetadata {
+    boolean enabled
+
+    WaveMetadata(Session session) {
+        if( session.config.wave as Map ) {
```

Comment:
Since the `WaveConfig` is declared in its own plugin it can be used the same approach for Fusion. This should be fine. Tho there's some convoluted logic that can impact this 

https://github.com/nextflow-io/nextflow/blob/4380403ef05e95b4b50691f2fe02b4272f022173/plugins/nf-wave/src/main/io/seqera/wave/plugin/WaveFactory.groovy#L38-L55

Reply:
> Since the `WaveConfig` is declared in its own plugin it can be used the same approach for Fusion

@pditommaso  can you clarify how I can use attributes of the `WaveConfig` plugin from within Nextflow `WaveMetadata` class? It is not clear to me.

Reply:
> Tho there's some convoluted logic that can impact this

Are you referring to the ability to disable Wave through the env variable `NXF_DISABLE_WAVE_SERVICE `?  
If so, couldn't we simply have it checked by the `WaveConfig` constructor, instead of `WaveFactory`?

Reply:
Sorry, I meant "it CANNOT be used the same approach for Fusion"

Reply:
Ah ok, it makes more sense!

Reply:
Ok, the current implementation  updates `wave.enabled` to `false` in the metadata,  when `NXF_DISABLE_WAVE_SERVICE` is set.

I would say it is fine, because it is the user that sets the variable and hence disables Wave, and the metadata correctly reflect this.

Or were you expecting another behaviour?

Reply:
It's fine